Source/WebCore/style/StyleAdjuster.cpp

```diff
@@ -301,7 +301,7 @@ void Adjuster::adjust(RenderStyle& style, const RenderStyle* userAgentAppearance
             }
 
             if (m_element->hasTagName(legendTag))
-                style.setEffectiveDisplay(DisplayType::Block);
+                style.setEffectiveDisplay(equivalentBlockDisplay(style));
```

Comment:
I think this puts us in line with Chrome, though the test says we shouldn't adjust style at all, not sure why we do it in the first place. I think @anttijk has more context on why we do it.

Reply:
Cool. We also have "display: block" in html.css similar to Chrome for UA stylesheet.